gips/data/core.py

```diff
@@ -357,6 +357,18 @@ def dates(cls, asset, tile, dates, days):
         dates = [dt for dt in datearr if days[0] <= int(dt.strftime('%j')) <= days[1]]
         return dates
 
+
+    @classmethod
+    def query_service(cls, asset, tile, date):
+        """Wrap call to emulate existing convention."""
+        if not hasattr(cls, 'query_provider'):
+            raise NotImplementedError('Query/fetch bifurcation not supported for ' + cls.__name__)
+        bn, url = cls.query_provider(asset, tile, date)
+        if (bn, url) == (None, None):
+            return []
+        return [{'basename': bn, 'url': url}]
```

Comment:
Add documentation along the lines of:

"""
Asset child classes may implement the `query_service` in 
any manner they see fit, so long as they return a `list` of `dict`s
for each file which must be fetched to provide the requested
`(asset, tile, date)`, which may be an empty list.

Some drivers defer this to a `query_provider` method which 
must return a couple of `(basename, url)`, or couple of `(None, None)`.
"""